ta1/output-schemas.py

```diff
@@ -0,0 +1,256 @@
+from pydantic import BaseModel, Field
+from typing import Optional, TypeVar
+from json import dumps
+from jsonschema2md import Parser
+
+import erdantic as erd
+
+Polygon = TypeVar("Polygon")
+Line = TypeVar("Line")
+Point = TypeVar("Point")
+Image = TypeVar("Image")
+WKT = TypeVar("WKT", bound=str)
+
+
+class GeologicAgeInformation(BaseModel):
+    """
+    Information about the geologic age of a map unit.
+    """
+
+    age_text: str = Field(
+        ..., description="Text representation of age extracted from legend"
+    )
+    t_interval: Optional[str | int] = Field(
+        ..., description="Youngest interval", examples=["Holocene", "Cretaceous"]
+    )
+    b_interval: Optional[str | int] = Field(
+        ..., description="Oldest interval", examples=["Mesozoic", "Neoproterozoic"]
+    )
+    t_age: Optional[int] = Field(..., description="Minimum age (in Ma)")
+    b_age: Optional[int] = Field(..., description="Maximum age (in Ma)")
+
+
+class MapUnit(GeologicAgeInformation):
+    id: int = Field(..., description="Internal ID")
+    name: str = Field(..., description="Geologic unit name extracted from legend")
+
+    color: str = Field(..., description="Color extracted from map/legend")
+    pattern: Optional[str] = Field(..., description="Pattern extracted from map/legend")
+    abbreviation: Optional[str] = Field(
+        ..., description="Abbreviation extracted from map/legend"
+    )
+    description: Optional[str] = Field(
+        ..., description="Description text extracted from legend"
+    )
+    lithology: list[str] = Field(
```

Comment:
Like age information, lithology information requires an external understanding
of geological descriptive language (i.e., which information in a legend text string represents
lithological attributes). This can be assessed with the help of an API (e.g., Macrostrat's
lithology definitions API) or other embedding model. Perhaps this should be more properly
grouped with the age information as a "geologic information" object.